examples/timestamping/backend/tests/config.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright 2020 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use exonum::runtime::{InstanceId, SnapshotExt};
+
+use exonum_merkledb::access::AccessExt;
+use exonum_rust_runtime::ServiceFactory;
+use exonum_supervisor::{ConfigPropose, Supervisor};
+use exonum_testkit::{TestKit, TestKitBuilder};
+use exonum_time::{MockTimeProvider, TimeServiceFactory};
+
+use std::time::SystemTime;
+
+use exonum_timestamping::{Config, TimestampingService};
+
+const TIME_SERVICE_ID: InstanceId = 102;
+const TIME_SERVICE_NAME: &str = "time";
+const SERVICE_ID: InstanceId = 103;
+const SERVICE_NAME: &str = "timestamping";
+const SECOND_TIME_SERVICE_ID: InstanceId = 104;
+const SECOND_TIME_SERVICE_NAME: &str = "time2";
+
+fn init_testkit(second_time_service: bool) -> (TestKit, MockTimeProvider) {
+    let mock_provider = MockTimeProvider::new(SystemTime::now().into());
+    let time_service = TimeServiceFactory::with_provider(mock_provider.clone());
+    let time_service_artifact = time_service.artifact_id();
+    let timestamping = TimestampingService;
+    let timestamping_artifact = timestamping.artifact_id();
+
+    let testkit = TestKitBuilder::validator()
+        .with_rust_service(Supervisor)
+        .with_artifact(Supervisor.artifact_id())
+        .with_instance(Supervisor::simple())
+        .with_rust_service(time_service)
+        .with_rust_service(timestamping)
+        .with_artifact(time_service_artifact.clone())
+        .with_instance(
+            time_service_artifact.into_default_instance(TIME_SERVICE_ID, TIME_SERVICE_NAME),
+        )
+        .with_artifact(timestamping_artifact.clone())
+        .with_instance(
+            timestamping_artifact
+                .into_default_instance(SERVICE_ID, SERVICE_NAME)
+                .with_constructor(Config {
+                    time_service_name: TIME_SERVICE_NAME.to_owned(),
+                }),
+        );
+
+    let testkit = if second_time_service {
+        let time_service = TimeServiceFactory::with_provider(mock_provider.clone());
+        let time_service_artifact = time_service.artifact_id();
+        testkit
+            .with_rust_service(time_service)
+            .with_artifact(time_service_artifact.clone())
```

Comment:
This also should be redundant: the time service artifact has already been added to the builder.